### PR TITLE
My profile

### PR DESCRIPTION
**Pull request by iamtarun1993**

### CODE REVIEW

File: src/app/core/services/transaction.service.ts [hunk 1]
```diff
@@ -69,4 +107,61 @@ export class TransactionService {
...
+          etxns = etxns.concat(restEtxns);
+          return etxns;
+        });
+      });
+    } else {
+      return this.apiService.get('/etxns', data);
+    }
+  }
+
+  getMyETxncInternalChunked(): Observable<any> {
```
Comment (Julias0) [inline src/app/core/services/transaction.service.ts:135]: we have a better way of doing this with v2 apis and rxjs - check getAllEouc method in orguserservice

File: src/app/fyle/my-profile/my-profile.page.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  async openCurrenySelectionModal(event) {
+    const modal = await this.modalController.create({
+      component: SelectCurrencyComponent
+    });
+
+    await modal.present();
+
+    const { data } = await modal.onWillDismiss();
+    if (data) {
+      this.currencies.some((currency) => {
```
Comment (Julias0) [inline src/app/fyle/my-profile/my-profile.page.ts:108]: use .find

File: src/app/fyle/my-profile/my-profile.page.ts [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+  toggleAutoExtraction() {
+    return this.orgUserSettingsService.post(this.orgUserSettings)
+    .pipe(
+      map((res) => {
+        console.log(res);
+        // Todo: Tracking service and disable toogle button
+      })
+    )
+    .subscribe();
```
Comment (Julias0) [inline src/app/fyle/my-profile/my-profile.page.ts:127]: pass in noop

File: src/app/fyle/my-profile/my-profile.page.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+  }
+
+  getAllCurrencyAndMatchPreferredCurrency() {
+    this.currencyService.getAllCurrenciesInList().pipe(
+      map((res) => {
+        console.log(res);
+        this.currencies = res;
+        this.currencies.some((currency) => {
```
Comment (Julias0) [inline src/app/fyle/my-profile/my-profile.page.ts:201]: find

File: src/app/core/services/transaction.service.ts (hunk 1 above)
Comment (iamtarun1993) [reply src/app/core/services/transaction.service.ts:135]: there is no getAllEouc in orguserservice in master branch

**Review by Julias0 — APPROVED** [review]

**Pull request closed without merging**